flash/image/classification/integrations/baal.py

```diff
@@ -0,0 +1,250 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from copy import deepcopy
+from functools import partial
+from typing import Any, Callable, Dict, Optional
+
+import numpy as np
+import torch
+from pytorch_lightning import LightningDataModule
+from pytorch_lightning.loops import Loop
+from pytorch_lightning.loops.fit_loop import FitLoop
+from pytorch_lightning.trainer.connectors.data_connector import _PatchDataLoader
+from pytorch_lightning.trainer.progress import Progress
+from pytorch_lightning.trainer.states import RunningStage, TrainerFn, TrainerStatus
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from torch.nn.modules.batchnorm import _BatchNorm
+from torch.utils.data import DataLoader
+
+import flash
+from flash import DataModule
+from flash.core.data.auto_dataset import BaseAutoDataset
+from flash.core.data.data_pipeline import DataPipeline
+from flash.core.utilities.imports import _BAAL_AVAILABLE, requires
+
+if _BAAL_AVAILABLE:
+    from baal.active.dataset import ActiveLearningDataset
+    from baal.active.heuristics import AbstractHeuristic, BALD
+
+
+def dataset_to_non_labelled_tensor(dataset: BaseAutoDataset) -> torch.tensor:
+    return torch.zeros(len(dataset))
+
+
+class ActiveLearningDataModule(LightningDataModule):
+    @requires("baal")
+    def __init__(
+        self,
+        labelled: Optional[DataModule] = None,
+        unlabelled: Optional[DataModule] = None,
+        heuristic: "AbstractHeuristic" = BALD(reduction=np.mean),
+        map_dataset_to_labelled: Optional[Callable] = dataset_to_non_labelled_tensor,
+    ):
+        self.labelled = labelled
+        self.unlabelled = unlabelled
+        self.heuristic = heuristic
+        self.map_dataset_to_labelled = map_dataset_to_labelled
+
+        if self.unlabelled:
+            raise MisconfigurationException("The unlabelled `datamodule` isn't support yet.")
```

Comment:
If the argument is not supported it might be cleaner to just remove it

Reply:
This is not supported, but I would like to work on this asap, so we support the production use case. The problem is that I need to label data ahah :)
I plan to integrate with LabelStudio, so I can do this from the UI.

Reply:
Ok, removed it and I will add it back when supported.